src/lightning/fabric/strategies/strategy.py

```diff
@@ -305,6 +306,39 @@ def teardown(self) -> None:
         self.accelerator.teardown()
         self.checkpoint_io.teardown()
 
+    def _unscale_params(self, optimizer):
+        scaler = getattr(self.precision, "scaler", None)
+        if scaler is not None:
+            if _optimizer_handles_unscaling(optimizer):
+                raise NotImplementedError("Gradient clipping is not implemented for optimizers handling the unscaling.")
+            scaler.unscale_(optimizer)
+
+    def clip_gradients_norm(
+        self,
+        optimizer,
+        max_norm: Union[float, int],
+        norm_type: Union[float, int] = 2.0,
+        error_if_nonfinite: bool = True,
+    ):
+        """Clip gradients by norm."""
+        from lightning.fabric.wrappers import _unwrap_objects
+
+        optimizer = _unwrap_objects(optimizer)
+        self._unscale_params(optimizer)
+        parameters = self.precision.main_params(optimizer)
+        return torch.nn.utils.clip_grad_norm_(
+            parameters, max_norm=max_norm, norm_type=norm_type, error_if_nonfinite=error_if_nonfinite
+        )
+
+    def clip_gradients_value(self, optimizer, clip_val: Union[float, int]):
```

Comment:
As briefly chatted offline, we may want to consider a single clip_gradients method, and the mode as a string. The strategy implementations would handle this similarly to the collectives, where sometimes not all possible inputs are supported.

Reply:
I know, this just wasn't added yet, but it will be :)